src/hooks/book/use-book-reviews.ts

```diff
@@ -16,6 +16,7 @@ import { Rating } from "@/lib/utils/Rating";
 export const useBookReviews = (isbn: string | undefined) => {
   const [reviews, setReviews] = useState<BookReview[]>([]);
   const [ratings, setRatings] = useState<BookReview[]>([]);
+  // Store the raw fraction value (0-1) for userRating
   const [userRating, setUserRating] = useState<number>(0);
```

Comment:
The state should hold the rating as a Rating object.

```suggestion
  const [userRating, setUserRating] = useState<Rating>(new Rating(0));
```